### PR TITLE
Update Anycubic i3 Mega

### PR DESCRIPTION
**Pull request by thinkyhead**
### Description

Update the i3 configuration to build successfully for unmodified Anycubic i3 MEGA

From #854 by @sobil

### Benefits

- Allows successful build 
- Provides optional z access pin allocation fix
- Fixes offset 

### Related Issues

https://github.com/MarlinFirmware/Marlin/pull/25071

### CODE REVIEW

File: config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h [hunk 1]
```diff
@@ -1736,7 +1745,7 @@
 #define Y_BED_SIZE 210
 
 // Travel limits (linear=mm, rotational=°) after homing, corresponding to endstop positions.
-#define X_MIN_POS -5
+#define X_MIN_POS 0
```
Comment (thinkyhead) [inline config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1747]: @sobil — Is it true that when X is homed the nozzle is over the bed in a usable location?

File: config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h [hunk 2]
```diff
@@ -1736,7 +1745,7 @@
...
 #define Y_MIN_POS 0
 #define Z_MIN_POS 0
 #define X_MAX_POS X_BED_SIZE
```
Comment (thinkyhead) [inline config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1750]: Is it true that this is the farthest to the right the X carriage can move to?

If your aim is to get the center coordinates 105, 105 to line up with the center of the bed, you can tweak other values and keep `X_MIN_POS` at -5 if the nozzle is left of the bed when homed.

File: config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h (hunk 1 above)
Comment (sobil) [reply config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1747]: Yes, that is correct.

See pic for reference: https://photos.app.goo.gl/dYX4kYrB5BLSSNr56

File: config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h (hunk 2 above)
Comment (sobil) [reply config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1750]: I'll be honest I didn't check the centre position. I was mostly concerned about getting the left edge of bed usable.

File: config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h (hunk 1 above)
Comment (sobil) [reply config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1747]: Hey @thinkyhead 

I've just realised that this shouldn't be changed. as some printers appear to be shipped with a slightly smaller bed than: 
image

Comment (sobil) [inline config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1747]:
```suggestion
#define X_MIN_POS -5
```

Can you please revert this.

File: config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h (hunk 2 above)
Comment (sobil) [reply config/examples/AnyCubic/i3 Mega/Trigorilla AVR/Configuration.h:1750]: Thanks for letting me double check. Mine seems to have an optional bed that's bigger than most.

**Pull request merged**